src/rgw/rgw_rest_conn.cc

```diff
@@ -180,17 +180,17 @@ int RGWRESTConn::get_obj(const rgw_user& uid, req_info *info /* optional */, rgw
   }
   map<string, string> extra_headers;
   if (info) {
-    map<string, string, ltstr_nocase>& orig_map = info->env->get_map();
+    const auto& orig_map = info->env->get_map();
 
     /* add original headers that start with HTTP_X_AMZ_ */
 #define SEARCH_AMZ_PREFIX "HTTP_X_AMZ_"
-    for (map<string, string>::iterator iter = orig_map.lower_bound(SEARCH_AMZ_PREFIX); iter != orig_map.end(); ++iter) {
-      const string& name = iter->first;
+    for (const auto& iter: orig_map) {
```

Comment:
hm this doesn't  look right, use the lower bound, and maybe #define -> constexpr?